dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05400LoadAppsSecrets.java

```diff
@@ -0,0 +1,71 @@
+package com.dotmarketing.startup.runonce;
+
+import com.dotcms.security.apps.AppSecrets;
+import com.dotcms.security.apps.AppsAPI;
+import com.dotcms.security.apps.AppsAPIImpl;
+import com.dotcms.security.apps.AppsUtil;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.startup.StartupTask;
+import com.google.common.annotations.VisibleForTesting;
+import com.liferay.portal.model.User;
+import io.vavr.control.Try;
+import java.io.File;
+import java.nio.file.DirectoryStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.security.Key;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class Task05400LoadAppsSecrets implements StartupTask {
+
+    private static final Pattern importFilePattern = Pattern.compile("^dotSecrets-import\\.([a-zA-Z0-9-_]+)", Pattern.CASE_INSENSITIVE);
+
+    static boolean matchesFileName(final String input) {
+        final Matcher matcher = importFilePattern.matcher(input);
+        return matcher.find();
+    }
+
+    final AppsAPI appsAPI = APILocator.getAppsAPI();
```

Comment:
We should not use API in the Upgrade Task, because cache and API can change between versions.

Reply:
I don't think we have another option here. This API writes directly into a file. it doesn't do any db hits or anything else. I removed all that so it could be used from here safely.

Reply:
agree

Reply:
I refactored my code so it could do the import without having to use the APi  @erickgonzalez